jupyter_scheduler/scheduler.py

```diff
@@ -18,149 +25,249 @@
     ListJobDefinitionsResponse,
     ListJobsQuery,
     ListJobsResponse,
+    Output,
     SortDirection,
     Status,
     UpdateJob,
     UpdateJobDefinition,
 )
 from jupyter_scheduler.orm import Job, JobDefinition, create_session
-from jupyter_scheduler.utils import (
-    compute_next_run_time,
-    create_output_filename,
-    resolve_path,
-    timestamp_to_int,
-)
+from jupyter_scheduler.utils import create_output_filename
 
 
-class BaseScheduler(ABC):
+class BaseScheduler(LoggingConfigurable):
     """Base class for schedulers. A default implementation
     is provided in the `Scheduler` class, but extension creators
     can provide their own scheduler by subclassing this class.
     By implementing this class, you will replace both the service
     API and the persistence layer for the scheduler.
     """
 
-    @abstractmethod
+    staging_path = Unicode(
+        help=_i18n(
+            """Full path to staging location, where output
+        files will be stored after job execution completes. This
+        could be a local or remote path including cloud storage.
+        Default value is jupyter data directory.
+        """
+        )
+    )
+
+    @default("staging_path")
+    def _default_staging_path(self):
+        return jupyter_data_dir()
```

Comment:
This should be under `jupyter_data_dir()/scheduler_staging_area` to avoid polluting `jupyter_data_dir()` with too many children.

Reply:
This comment is blocking.

Reply:
Latest changes look good.